lib/ruby_lsp/requests/formatting.rb

```diff
@@ -33,6 +33,8 @@ def run
         formatted_text = formatted_file
         return unless formatted_text
 
+        return if formatted_text == @document.source
```

Comment:
A slight optimization here for large documents might be to check the size of each of these strings before comparing them byte-for-byte:
```suggestion
        return if formatted_text.size == @document.source.size && formatted_text == @document.source
```
This should speed up the cases where the formatted text is different from the original by catching the difference in the sizes as early as possible.

Reply:
Very nice!